### PR TITLE
[BE] Stop using deprecated APIs in mkldnn_pattern_matcher

### PR DESCRIPTION
**Pull request by malfet**
This should fix
```
/var/lib/jenkins/workspace/test/inductor/test_mkldnn_pattern_matcher.py:157: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by kit1980 — APPROVED** [review]

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def cal_conv_generated_kernel_number(mod, input, dtype, dim=4):
     if dtype == torch.float32:
         maybe_autocast = contextlib.nullcontext()
     else:
-        maybe_autocast = torch.cpu.amp.autocast(dtype=dtype)
+        maybe_autocast = torch.amp.autocast('cpu', dtype=dtype)
```
Comment (malfet) [inline test/inductor/test_mkldnn_pattern_matcher.py:107]:
```suggestion
        maybe_autocast = torch.amp.autocast("cpu", dtype=dtype)
```

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -156,13 +156,13 @@ def _test_common(
             check_autocast == torch.bfloat16
             and torch.ops.mkldnn._is_mkldnn_bf16_supported()
         ):
-            maybe_autocast = torch.cpu.amp.autocast(dtype=torch.bfloat16)
+            maybe_autocast = torch.amp.autocast('cpu', dtype=torch.bfloat16)
```
Comment (malfet) [inline test/inductor/test_mkldnn_pattern_matcher.py:159]:
```suggestion
            maybe_autocast = torch.amp.autocast("cpu", dtype=torch.bfloat16)
```

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 3]
```diff
@@ -156,13 +156,13 @@ def _test_common(
...
             atol, rtol = 1e-2, 1e-2
         elif (
             check_autocast == torch.float16
             and torch.ops.mkldnn._is_mkldnn_fp16_supported()
         ):
-            maybe_autocast = torch.cpu.amp.autocast(dtype=torch.float16)
+            maybe_autocast = torch.amp.autocast('cpu', dtype=torch.float16)
```
Comment (malfet) [inline test/inductor/test_mkldnn_pattern_matcher.py:165]:
```suggestion
            maybe_autocast = torch.amp.autocast("cpu", dtype=torch.float16)
```

**Comment by kit1980** [comment]
Added TorchFix a rule for this: https://github.com/pytorch-labs/torchfix/pull/87

**Pull request closed without merging**